README.md
# Plugin plxMySocialNetwork pour PluXml
plxMySocialNetwork (fork de aplxSocialImg) : partager vos articles sur les réseaux sociaux

![demo](https://cloud.githubusercontent.com/assets/13441278/10242180/527d3188-68ef-11e5-9ae4-cd4ebb8be4d2.png)



## Fonctionnalités

* Affiche un lien pour partager un article sur les réseaux sociaux
* Réseaux sociaux: Facebook, Twitter, Google+, LinkedIn, Diaspora et Courriel
* Ajouter d'autres liens vers des réseaux sociaux
* Partager les articles et/ou les pages statiques
* Affichage des liens en texte ou boutons via font-awesome (à valider dans les paramètres)
* Pas de code tiers


## Insérer plxMySocialNetwork le thème utilisé

Dans le fichier article.php,article-full-width.php et/ou static.php, static-full-width.php de votre thème, ajoutez la ligne suivante à l'endroit où vous souhaitez afficher les boutons.


    <?php eval($plxShow->callHook('MySocialNetwork')) ?>


## Modifier le CSS

Paramètres > Plugins > menu "Plugins actifs" > plugin "MySocialNetwork" > menu "Code css" > champ "Contenu fichier css site"


avec font-awesome:

![font-awesome](https://cloud.githubusercontent.com/assets/13441278/10242181/527e8ae2-68ef-11e5-990c-4f2145616495.png)
  
## Modifier le nombre de liens maximum

dans le fichier **config.php** modifier la valeur de *$maxItems*:

    // nb de liens max
    $maxItems= 10; 
  
## Evolutions possible

* modifier l'ordre des liens
* définir le nb de liens directement depuis le panneau de configuration du plugin

![icon](https://cloud.githubusercontent.com/assets/13441278/9706299/84f5fdc4-54e1-11e5-96ef-d5ad697a1e32.png)


changelog.txt
## Version 1.5 (02/10/2015) ##
correction bug flux RSS
suppression ob_start(); 
ajout rawurlencode($url)
modification service Delicious


## Version 1.4 (26/09/2015) ##
modification avec un seul hook pour gérer les articles et pages statiques
ajout CSS avec images
mise à jour des traductions
ajout de nouveaux services


## Version 1.3 (25/09/2015) ##
Suppression du nb de liens maximum 
mise à jour des traductions
modification du formulaire de config

## Version 1.2 (22/09/2015) ##
Ajout de la traduction française

## Version 1.1 (21/09/2015) ##
Ajout d'un formulaire pour ajouter les réseaux sociaux
Possibiliter d'activer/désactiver individuellement chacun des liens
Modification CSS par défaut
Ajout de réseaux sociaux à l'activation

## Version 1.0 (06/09/2015) ##
version initiale


config.php
<?php if(!defined('PLX_ROOT')) exit; ?>


<?php

// nb de liens max
$maxItems= 12; 


# Control du token du formulaire
plxToken::validateFormToken($_POST);


if(!empty($_POST)) {
	
  for  ($i = 1; $i <= $maxItems; $i++) 
		{     
			$plxPlugin->setParam($i.'-param1', $_POST[$i.'-param1'], 'cdata');
			$plxPlugin->setParam($i.'-param2', $_POST[$i.'-param2'], 'string');
			$plxPlugin->setParam($i.'-param3', $_POST[$i.'-param3'], 'cdata');
			$plxPlugin->setParam($i.'-param4', $_POST[$i.'-param4'], 'cdata');
			$plxPlugin->setParam($i.'-param5', $_POST[$i.'-param5'], 'cdata');     
		}


	$plxPlugin->setParam('mail-param1', $_POST['mail-param1'], 'cdata');
	$plxPlugin->setParam('mail-param2', $_POST['mail-param2'], 'string');
	$plxPlugin->setParam('mail-param3', $_POST['mail-param3'], 'cdata');
	$plxPlugin->setParam('mail-param4', $_POST['mail-param4'], 'cdata');
	$plxPlugin->setParam('mail-param5', $_POST['mail-param5'], 'cdata');
	$plxPlugin->setParam('mail-param8', $_POST['mail-param8'], 'cdata');
	$plxPlugin->setParam('mail-param9', $_POST['mail-param9'], 'cdata');
	$plxPlugin->setParam('mail-param0', $_POST['mail-param0'], 'cdata');	

	$plxPlugin->setParam('maxItems', $_POST['maxItems'], 'numeric');
	$plxPlugin->setParam('share-article', $_POST['share-article'], 'cdata');
	$plxPlugin->setParam('share-page', $_POST['share-page'], 'cdata');

$plxPlugin->saveParams();
header('Location: parametres_plugin.php?p=plxMySocialNetwork');
exit;
}
?>


<div id="form">
<h2><?php  $plxPlugin->lang('L_MANAGE_PLUGIN') ?></h2>
<form class="inline-form" action="parametres_plugin.php?p=plxMySocialNetwork" method="post" id="form_test">
	
	<div class="scrollable-table">
	<table class="table full-width">
	<caption><?php  $plxPlugin->lang('L_SOCIAL_NETWORK') ?></caption>
		<thead>
			<tr>
				<th><?php  $plxPlugin->lang('L_NAME') ?></th>
				<th><?php  $plxPlugin->lang('L_ACTIVE') ?> <a class="help" title="<?php  $plxPlugin->lang('L_ACTIVE_HELP') ?>">&nbsp;</a></th>
				<th><?php  $plxPlugin->lang('L_URL') ?> <a class="help" title="<?php  $plxPlugin->lang('L_URL_HELP') ?>">&nbsp;</a></th>
				<th><?php  $plxPlugin->lang('L_DESCRIPTION') ?> <a class="help" title="<?php  $plxPlugin->lang('L_DESCRIPTION_HELP') ?>">&nbsp;</a></th>
				<th><?php  $plxPlugin->lang('L_CLASS') ?> <a class="help" title="<?php  $plxPlugin->lang('L_CLASS_HELP') ?>">&nbsp;</a></th>
			</tr>
		</thead>
		<tbody>
				<?php 	
				$newItem=1;
				for  ($i = 1; $i <= $maxItems; $i++) 
				{    
				  if (($plxPlugin->getParam($i.'-param3')) != "" ){ echo "	<tr>";
																					echo "<td>"; plxUtils::printInput($i.'-param1',$plxPlugin->getParam($i.'-param1'),'text','20-50') ; echo "</td>";
																					echo "<td>"; plxUtils::printSelect($i.'-param2',array('0'=>'Masquer','1'=>'Afficher'), $plxPlugin->getParam($i.'-param2')); echo "</td>";
																					echo "<td>"; plxUtils::printInput($i.'-param3',$plxPlugin->getParam($i.'-param3'),'text','50-255') ; echo "</td>";
																					echo "<td>"; plxUtils::printInput($i.'-param4',$plxPlugin->getParam($i.'-param4'),'text','35-100') ; echo "</td>";
																					echo "<td>"; plxUtils::printInput($i.'-param5',$plxPlugin->getParam($i.'-param5'),'text','10-20') ; echo "</td>";
																	echo "</tr>" ;} 
																	
				  elseif ($newItem==1){  $newItem=0;  			echo "	<tr>";
																					echo "<td>"; plxUtils::printInput($i.'-param1',$plxPlugin->getParam($i.'-param1'),'text','20-50') ; echo "</td>";
																					echo "<td>"; plxUtils::printSelect($i.'-param2',array('0'=>'Masquer','1'=>'Afficher'), $plxPlugin->getParam($i.'-param2')); echo "</td>";
																					echo "<td>"; plxUtils::printInput($i.'-param3',$plxPlugin->getParam($i.'-param3'),'text','50-255') ; echo "</td>";
																					echo "<td>"; plxUtils::printInput($i.'-param4',$plxPlugin->getParam($i.'-param4'),'text','35-100') ; echo "</td>";
																					echo "<td>"; plxUtils::printInput($i.'-param5','share-alt','text','10-20') ; echo "</td>";
																	echo "</tr>" ;} 
				} 
				?>

			<tr><td colspan="5"><div id="help">

			<p><?php  $plxPlugin->lang('L_PATTERN_URL') ?><br />
			   <?php  $plxPlugin->lang('L_FORM_URL') ?>: <code>https://twitter.com/share?url={$url}&amp;text={$title}</code><br />
			   <?php  $plxPlugin->lang('L_EXEMPLE_URL') ?>: <code>https://twitter.com/share?url=https://blog.niqnutn.com/index.php?article12&text=ArticleAPartager</code>
			</p>
		
			</div></td></tr>
		

		</tbody>
		
	</table>
	</div>
	
	<div class="scrollable-table">
	<table class="table full-width">
	<caption><?php  $plxPlugin->lang('L_EMAIL') ?></caption>
		<thead>
			<tr>
				<th><?php  $plxPlugin->lang('L_NAME') ?></th>
				<th><?php  $plxPlugin->lang('L_ACTIVE') ?> <a class="help" title="<?php  $plxPlugin->lang('L_ACTIVE_HELP') ?>">&nbsp;</a></th>
				<th><?php  $plxPlugin->lang('L_URL') ?> <a class="help" title="<?php  $plxPlugin->lang('L_URL_HELP') ?>">&nbsp;</a></th>
				<th><?php  $plxPlugin->lang('L_DESCRIPTION') ?> <a class="help" title="<?php  $plxPlugin->lang('L_DESCRIPTION_HELP') ?>">&nbsp;</a></th>
				<th><?php  $plxPlugin->lang('L_CLASS') ?> <a class="help" title="<?php  $plxPlugin->lang('L_CLASS_HELP') ?>">&nbsp;</a></th>
			</tr>
		</thead>	
		<tbody>	
			<tr>
				<td><?php plxUtils::printInput('mail-param1',$plxPlugin->getParam('mail-param1'),'text','20-50') ?></td>
				<td><?php plxUtils::printSelect('mail-param2',array('0'=>'Masquer','1'=>'Afficher'), $plxPlugin->getParam('mail-param2')) ?></td>
				<td><?php plxUtils::printInput('mail-param3','mailto:'.$plxPlugin->getParam('mail-param8').'?subject='.$plxPlugin->getParam('mail-param9').'&amp;body='.$plxPlugin->getParam('mail-param0'),'text','50-255', true)  ?></td>
				<td><?php plxUtils::printInput('mail-param4',$plxPlugin->getParam('mail-param4'),'text','35-100') ?></td>
				<td><?php plxUtils::printInput('mail-param5',$plxPlugin->getParam('mail-param5'),'text','10-20') ?></td>
			</tr>
			<tr>
				<td ><?php  $plxPlugin->lang('L_EMAIL_SUBJECT') ?></td>
				<td colspan="4"><?php plxUtils::printInput('mail-param9',$plxPlugin->getParam('mail-param9'),'text','80-255') ?></td>
			</tr>
			<tr>
				<td ><?php  $plxPlugin->lang('L_EMAIL_BODY') ?></td>
				<td colspan="4"><?php plxUtils::printInput('mail-param0',$plxPlugin->getParam('mail-param0'),'text','80-255') ?></td>
			</tr>
			<tr>
				<td ><?php  $plxPlugin->lang('L_EMAIL_RECIPIENT') ?></td>
				<td colspan="4"><?php plxUtils::printInput('mail-param8',$plxPlugin->getParam('mail-param8'),'text','80-255') ?></td>
			</tr>
		</tbody>
	</table>
	</div>
	
	<div class="scrollable-table">	
	<table class="table full-width">
	<caption>Autres param&egrave;tres</caption>
		<thead>
			<tr>
				<th><?php  $plxPlugin->lang('L_DESCRIPTION') ?></th>
				<th><?php  $plxPlugin->lang('L_VALUE') ?></th>

			</tr>
		</thead>	
		<tbody>	
			<tr>
				<td><?php $plxPlugin->lang('L_NB_ITEMS') ?></td>
				<td><?php plxUtils::printInput('maxItems',$maxItems,'numeric','2-2', true) ?></td>
			</tr>
			<tr>
				<td><?php $plxPlugin->lang('L_LABEL_SHARE_ARTICLE') ?></td>
				<td><?php plxUtils::printInput('share-article',$plxPlugin->getParam('share-article'),'text','35-100') ?></td>
			</tr>
			<tr>
				<td><?php $plxPlugin->lang('L_LABEL_SHARE_STATIC') ?></td>
				<td><?php plxUtils::printInput('share-page',$plxPlugin->getParam('share-page'),'text','35-100') ?></td>
			</tr>
		</tbody>
	</table>
	</div>
		
<p class="in-action-bar"><?php echo plxToken::getTokenPostMethod() ?><input type="submit" name="submit" value="<?php  $plxPlugin->lang('L_SAVE_CONFIG') ?>" /></p>

</div>







lang/fr.php
<?php
	$LANG = array(

# config.php
		'L_PLUGIN_TITLE'	=>	'plxMySocialNetwork',
		'L_MANAGE_PLUGIN'	=>	'Gestion du plugin MySocialNetwork',
		'L_SOCIAL_NETWORK'  =>  'Réseaux sociaux',
		'L_NAME'			=>  'Nom',
		'L_ACTIVE'		=>	'Active',
		'L_ACTIVE_HELP'	=>	'Activer / Désactiver le lien',
		'L_URL'			=>	'URL',
		'L_URL_HELP'		=>	'URL complète du lien pour partager l\'article / page statique',
		'L_DESCRIPTION'	=>	'Description',
		'L_DESCRIPTION_HELP'=>	'Description qui sera afficher pour le lien et infobulle',
		'L_CLASS'			=>	'CSS class',
		'L_CLASS_HELP'		=>	'Nom de la classe utilisée pour le CSS',
		'L_EMAIL'			=>	'Courriel',
		'L_EMAIL_SUBJECT'	=>	'Sujet',
		'L_EMAIL_BODY'		=>	'Corps du message',
		'L_EMAIL_RECIPIENT'	=>	'Destinataire',
		'L_HELP'			=> 	'Aide',
		'L_NB_ITEMS'		=> 	'Nombre maximum de liens',
		'L_VALUE'			=> 	'Valeur',
		'L_SAVE_CONFIG'	=>	'Enregistrer',
		'L_PATTERN_URL'	=>	'Remplacer l\'url de la page à partager par {$url} et son titre par {$title}.',
		'L_FORM_URL'		=>	'URL à indiquée dans le formulaire',
		'L_EXEMPLE_URL'	=>	'URL qui sera affichée',
		'L_LABEL_SHARE_ARTICLE'			=> 	'Libellé pour partager l\'article',
		'L_LABEL_SHARE_STATIC'			=> 	'Libellé pour partager la page',
		'L_LABEL_SHARE_LINKS'			=> 	'Libellé pour partager les liens',
		
# plxMySocialNetwork
		'L_SHARE_ARTICLE'			=>	'Partager l\'article: ',
		'L_SHARE_STATIC'			=>	'Partager la page: '

	);
?>


lang/fr-help.php
<h2>Aide</h2>

Pour afficher les liens des r&eacute;seaux sociaux Facebook, Twitter, Google+1, LinkedIn, Diaspora, Courriel, ...

<h3 style="font-size:1.3em;font-weight:bold;padding:10px 0 10px 0">Insérer les liens dans le thème utilisé</h3>
<p>
Dans le fichier <strong>article.php,article-full-width.php et/ou static.php, static-full-width.php</strong> de votre thème, ajoutez la ligne suivante à l'endroit où vous souhaitez afficher les boutons.
</p>
<pre style="padding-left:40px"><code>
&lt;?php eval($plxShow->callHook('MySocialNetwork')) ?&gt;
</code></pre>


<h3 style="font-size:1.3em;font-weight:bold;padding:10px 0 10px 0">Modification affichage des boutons</h3>
<p>
L'affichage des boutons peut être modifié, en personnalisant la classe <strong>.social-network </strong> en allant sur l'écran d'administration:
<br /><br />
Paramètres > Plugins > menu "Plugins actifs" > plugin "MySocialNetwork" > menu "Code css" > champ "Contenu fichier css site"
</p>

<p>
Cliquez sur le bouton "Sauvegarder le fichier" pour enregistrer les modifications.
</p>



<h3 style="font-size:1.3em;font-weight:bold;padding:10px 0 10px 0">Modifier le nombre maximum de liens</h3>

<p>
dans le fichier <strong>config.php</strong> modifier la valeur de <em>$maxItems</em>:
<pre  style="padding-left:40px"><code>// nb de liens max
$maxItems= 10; 
</code></pre>
 </p>


aplxSocialImg.php
<?php
	/*
	*	Plugin aplxSocialImg par aruhuno
	*	D'après l'idée de Websyys (plxSocialButtons)
	*	
	*	Fork 29 octobre 2013
	*	Update 15 février 2014
	*/

	class aplxSocialImg extends plxPlugin {

		/**
		* Constructeur de la classe
		*
		* @param	default_lang	langue par défaut
		* @return	stdio
		* @author	aruhuno
		**/
		public function __construct($default_lang) {

			# appel du constructeur de la classe plxPlugin
			parent::__construct($default_lang);

			# hook
			$this->addHook('ThemeEndHead', 'ThemeEndHead');
			$this->addHook('SocialImg', 'SocialImg');
		}
		
		/**
		* Méthode qui ajoute le code css dans la partie <head>
		*
		* @return	stdio
		* @author	aruhuno
		**/
		public function ThemeEndHead() {
			echo '<link rel="stylesheet" href="plugins/aplxSocialImg/aplxSocialImg.css" media="screen"/>';
		}

		/**
		* Méthode qui affiche les boutons sociaux
		*
		* @return	stdio
		* @author	aruhuno
		**/
		public function SocialImg() {
			$plxShow = plxShow::getInstance(); // permet l'utilisation de plxShow
			$onclick = "onclick=\"javascript:window.open(this.href, '', 'menubar=no,toolbar=no,resizable=yes,scrollbars=yes,height=600,width=600');return false;\"";
			
			/* Gestion du titre */
			ob_start();
			$plxShow->artTitle();
			$title = ob_get_clean();
			$title = urldecode($title); // décodage du titre
			$title = str_replace(' ', '+',$title); // mise en forme
			
			/* Récupération de l'URL */
			$url = "?article".$plxShow->artId();
			$url = $plxShow->plxMotor->urlRewrite($url);
			
			/* Affichage */
			echo "<div class='social'>";
				echo "<ul>";
					// Commentaires (si actifs)
					if($plxShow->plxMotor->plxRecord_arts->f('allow_com') AND $plxShow->plxMotor->aConf['allow_com']) {
						/* Gestion du nombre de commentaires */
						ob_start();
						$plxShow->artNbCom();
						$nbcom = ob_get_clean();
						$nbcom = strip_tags($nbcom);
						
						echo "<li><a class='comment' href='{$url}#comments'><span>{$nbcom}</span></a></li>";
					}
					// Google+
					echo "<li><a class='gplus' href='https://plus.google.com/share?url={$url}' {$onclick}><span>Partager sur Google+</span></a></li>";
					// Facebook
					echo "<li><a class='facebook' href='https://www.facebook.com/sharer/sharer.php?u={$url}' {$onclick}><span>Partager sur Facebook</span></a></li>";
					// Twitter
					echo "<li><a class='twitter' href='https://twitter.com/share?url={$url}&amp;text={$title}' {$onclick}><span>Partager sur Twitter</span></a></li>";
					// Linkedin
					echo "<li><a class='linkedin' href='https://www.linkedin.com/shareArticle?mini=true&amp;url={$url}&amp;title={$title}' {$onclick}><span>Partager sur LinkedIn</span></a></li>";
				echo "</ul>";
			echo "</div>";
			
			return;
		}
	}
?>


plxMySocialButtons.php
<?php
 /**
 * Plugin plxMySocialNetwork par nIQnutn
 *
 * D'après le plugin aplxSocialImg par aruhuno
 * 
 * Fork 06 september 2015
 * Update 06 september 2015
 *
 **/
class plxMySocialNetwork extends plxPlugin {


	/**
	 * Constructeur de la classe
	 *
	 * @param	default_lang	langue par défaut
	 * @return	stdio
	 * @author	nIQnutn
	 **/
	public function __construct($default_lang) {

		# appel du constructeur de la classe plxPlugin (obligatoire)
		parent::__construct($default_lang);

		# déclaration des hooks
		$this->addHook('MySocialNetworkArticle', 'MySocialNetworkArticle');
		$this->addHook('MySocialNetworkStatic', 'MySocialNetworkStatic');
	}



	/**
	 * Méthode qui affiche les boutons sociaux 
	 *
	 * @return	stdio
	 * @author	Stephane F, aruhuno, nIQnutn
	 **/
	public function MySocialNetworkArticle() {
		
		
		
		$plxShow = plxShow::getInstance(); // permet l'utilisation de plxShow
		$onclick = "onclick=\"javascript:window.open(this.href, '', 'menubar=no,toolbar=no,resizable=yes,scrollbars=yes');return false;\"";
				

		/* Récupération de l'URL */
		$url = "?article".$plxShow->artId();
		$url = $plxShow->plxMotor->urlRewrite($url);
	


		/* Gestion du titre */
		ob_start();
		$plxShow->artTitle();
		$title = ob_get_clean();



		echo "<div class=\"social-network\"><span class=\"share\">Partager cet article: </span><ul>";
	
		// Twitter
		echo "<li><a  title=\"Partager sur Twitter\"  class=\"twitter\" href=\"https://twitter.com/share?url={$url}&text={$title}\" {$onclick} target=\"_blank\" rel=\"nofollow\" ><span>Partager sur Twitter</span></a></li>";								
		//Facebook
		echo "<li><a title=\"Partager sur Facebook\" class=\"facebook\" href=\"https://www.facebook.com/sharer/sharer.php?u={$url}&t={$title}\" {$onclick} target=\"_blank\" rel=\"nofollow\" ><span>Partager sur Facebook</span></a></li>";
		// Google+
		echo "<li><a title=\"Partager sur Google+\" class=\"google-plus\" href=\"https://plus.google.com/share?url={$url}\" {$onclick}  target=\"_blank\" rel=\"nofollow\"  ><span>Partager sur Google+</span></a></li>";
		// Linkedin
		echo "<li><a title=\"Partager sur LinkedIn\" class=\"linkedin\" href=\"https://www.linkedin.com/shareArticle?mini=true&url={$url}&title={$title}\" {$onclick}  target=\"_blank\" rel=\"nofollow\"  ><span>Partager sur LinkedIn</span></a></li>";
		// Diaspora
		echo "<li><a title=\"Partager sur Diaspora\" class=\"diaspora\" href=\"https://share.diasporafoundation.org/?title={$title}&url={$url}\" {$onclick}  target=\"_blank\" rel=\"nofollow\"  ><span>Partager sur Diaspora</span></a></li>";
		// Courriel
		echo "<li><a title=\"Partager par courriel\" class=\"courriel\" href=\"mailto:?subject={$title}&body={$title}: {$url}\"   target=\"_blank\" rel=\"nofollow\"  ><span>Partager par courriel</span></a></li>";


		echo "</ul></div>";
	
	}

	/**
	 * Méthode qui affiche les boutons sociaux sur une page statique
	 *
	 * @return	stdio
	 * @author	Stephane F, aruhuno, nIQnutn
	 **/
	public function MySocialNetworkStatic() {
				
		$plxShow = plxShow::getInstance(); // permet l'utilisation de plxShow
		$onclick = "onclick=\"javascript:window.open(this.href, '', 'menubar=no,toolbar=no,resizable=yes,scrollbars=yes');return false;\"";
				
		/* Récupération de l'URL */
		$url = "?static".$plxShow->staticId();
		$url = $plxShow->plxMotor->urlRewrite($url);

		/* Gestion du titre */
		ob_start();
		$plxShow->staticTitle();
		$title = ob_get_clean();



		echo "	<div class=\"social-network\"><ul>";
		echo "<span class=\"share\">Partager cette page: </span> ";

		// Twitter
		echo "<li><a  title=\"Partager sur Twitter\"  class=\"twitter\" href=\"https://twitter.com/share?url={$url}&text={$title}\" {$onclick} target=\"_blank\" rel=\"nofollow\" ><span>Partager sur Twitter</span></a></li>";								
		//Facebook
		echo "<li><a title=\"Partager sur Facebook\" class=\"facebook\" href=\"https://www.facebook.com/sharer/sharer.php?u={$url}&t={$title}\" {$onclick} target=\"_blank\" rel=\"nofollow\" ><span>Partager sur Facebook</span></a></li>";
		// Google+
		echo "<li><a title=\"Partager sur Google+\" class=\"google-plus\" href=\"https://plus.google.com/share?url={$url}\" {$onclick}  target=\"_blank\" rel=\"nofollow\"  ><span>Partager sur Google+</span></a></li>";
		// Linkedin
		echo "<li><a title=\"Partager sur LinkedIn\" class=\"linkedin\" href=\"https://www.linkedin.com/shareArticle?mini=true&url={$url}&title={$title}\" {$onclick}  target=\"_blank\" rel=\"nofollow\"  ><span>Partager sur LinkedIn</span></a></li>";
		// Diaspora
		echo "<li><a title=\"Partager sur Diaspora\" class=\"diaspora\" href=\"https://share.diasporafoundation.org/?title={$title}&url={$url}\" {$onclick}  target=\"_blank\" rel=\"nofollow\"  ><span>Partager sur Diaspora</span></a></li>";
		// Courriel
		echo "<li><a title=\"Partager par courriel\" class=\"courriel\" href=\"mailto:?subject={$title}&body={$title}: {$url}\"   target=\"_blank\" rel=\"nofollow\"  ><span>Partager par courriel</span></a></li>";


		echo "</ul></div>";

	}
}
?>


plxMySocialNetwork.php
<?php
 /**
 * Plugin plxMySocialNetwork by nIQnutn
 * Update: 02 october 2015
 * Version: 1.5
 * 
 * Forked from aplxSocialImg by aruhuno on 06 september 2015
 **/
class plxMySocialNetwork extends plxPlugin {


	/**
	 * Constructeur de la classe 
	 *
	 * @param	default_lang	langue par défaut
	 * @return	stdio
	 * @author	nIQnutn
	 **/
	public function __construct($default_lang) {

		# appel du constructeur de la classe plxPlugin (obligatoire)
		parent::__construct($default_lang);
		
		# limite l'accès à l'écran d'administration du plugin
		$this-> setConfigProfil(PROFIL_ADMIN);		


		# déclaration des hooks
		$this->addHook('MySocialNetwork', 'MySocialNetwork');
	}


	 /**
	 * Méthode utilisée à l'activation du plugin
	 * initialise les réseaux sociaux par défaut 
	 * 
	 * @author	 nIQnutn
	 **/
	public function OnActivate() {

			$this->setParam('maxItems', '10', 'string');
			$this->setParam('share-article', 'Partager l\'article: ', 'cdata');
			$this->setParam('share-page', 'Partager la page: ', 'cdata');
			$this->setParam('share-links', 'Retrouvez moi sur:', 'cdata');

			$this->setParam('mail-param1', 'Courriel', 'cdata');
			$this->setParam('mail-param2', '1', 'string');
			$this->setParam('mail-param3', 'mailto:?subject={$title}&amp;body={$title}:&nbsp;{$url}', 'cdata');
			$this->setParam('mail-param4', 'Partager par courriel', 'cdata');
			$this->setParam('mail-param5', 'courriel', 'cdata');
			$this->setParam('mail-param9', '{$title}', 'cdata');
			$this->setParam('mail-param0', '{$title}:&nbsp;{$url}', 'cdata');
			
			$this->setParam('1-param1', 'Twitter', 'cdata');
			$this->setParam('1-param2', '1', 'string');
			$this->setParam('1-param3', 'https://twitter.com/share?url={$url}&text={$title}', 'cdata');
			$this->setParam('1-param4', 'Partager sur Twitter', 'cdata');
			$this->setParam('1-param5', 'twitter', 'cdata');

			$this->setParam('2-param1', 'Facebook', 'cdata');
			$this->setParam('2-param2', '1', 'string');
			$this->setParam('2-param3', 'https://www.facebook.com/sharer/sharer.php?u={$url}&t={$title}', 'cdata');
			$this->setParam('2-param4', 'Partager sur Facebook', 'cdata');
			$this->setParam('2-param5', 'facebook', 'cdata');

			$this->setParam('3-param1', 'Google+', 'cdata');
			$this->setParam('3-param2', '1', 'string');
			$this->setParam('3-param3', 'https://plus.google.com/share?url={$url}', 'cdata');
			$this->setParam('3-param4', 'Partager sur Google+', 'cdata');
			$this->setParam('3-param5', 'google-plus', 'cdata');
			
			$this->setParam('4-param1', 'LinkedIn', 'cdata');
			$this->setParam('4-param2', '1', 'string');
			$this->setParam('4-param3', 'https://www.linkedin.com/shareArticle?mini=true&amp;url={$url}&amp;title={$title}', 'cdata');
			$this->setParam('4-param4', 'Partager sur LinkedIn', 'cdata');
			$this->setParam('4-param5', 'linkedin', 'cdata');

			$this->setParam('5-param1', 'Diaspora', 'cdata');
			$this->setParam('5-param2', '1', 'string');
			$this->setParam('5-param3', 'https://share.diasporafoundation.org/?title={$title}&amp;url={$url}', 'cdata');
			$this->setParam('5-param4', 'Partager sur Diaspora', 'cdata');
			$this->setParam('5-param5', 'diaspora', 'cdata');

			$this->setParam('6-param1', 'Reddit', 'cdata');
			$this->setParam('6-param2', '1', 'string');
			$this->setParam('6-param3', 'http://www.reddit.com/submit?url={$url}&title={$title}', 'cdata');
			$this->setParam('6-param4', 'Partager sur Reddit', 'cdata');
			$this->setParam('6-param5', 'reddit', 'cdata');
			
			$this->setParam('7-param1', 'StumbleUpon', 'cdata');
			$this->setParam('7-param2', '1', 'string');
			$this->setParam('7-param3', 'http://www.stumbleupon.com/submit?url={$url}&title={$title}', 'cdata');
			$this->setParam('7-param4', 'Partager sur StumbleUpon', 'cdata');
			$this->setParam('7-param5', 'stumbleupon', 'cdata');	

			$this->setParam('8-param1', 'Tumblr', 'cdata');
			$this->setParam('8-param2', '1', 'string');
			$this->setParam('8-param3', 'http://www.tumblr.com/share?v=3&u={$url}&t={$title}', 'cdata');
			$this->setParam('8-param4', 'Partager sur Tumblr', 'cdata');
			$this->setParam('8-param5', 'tumblr', 'cdata');	

			$this->setParam('9-param1', 'Delicious', 'cdata');
			$this->setParam('9-param2', '1', 'string');
			$this->setParam('9-param3', 'https://delicious.com/save?v=5?url={$url}&title={$title}', 'cdata');
			$this->setParam('9-param4', 'Partager sur Delicious', 'cdata');
			$this->setParam('9-param5', 'delicious', 'cdata');	
						
			$this->setParam('10-param1', 'Pocket', 'cdata');
			$this->setParam('10-param2', '1', 'string');
			$this->setParam('10-param3', 'https://getpocket.com/save?url={$url}&title={$title}', 'cdata');
			$this->setParam('10-param4', 'Partager sur Pocket', 'cdata');
			$this->setParam('10-param5', 'pocket', 'cdata');	
									
		$this->saveParams();		
		
	}
	/**
	* Méthode qui affiche les liens
	*
	* @author	Stephane F, aruhuno, nIQnutn
	**/
	public function MySocialNetwork() {

		$plxShow = plxShow::getInstance(); // permet l'utilisation de plxShow
		$onclick = "onclick=\"javascript:window.open(this.href, '', 'menubar=no,toolbar=no,resizable=yes,scrollbars=yes');return false;\"";
	
		$url="";
		$title="";
		
		// Test article ?? page statique
		if($plxShow->mode()=='article')
			{
				/* Récupération de l'URL */
				$url = "?article".$plxShow->artId();
				$url = $plxShow->plxMotor->urlRewrite($url);
        			$url = rawurlencode($url);
            
				/* Gestion du titre */
				$plxShow->artTitle();
				$title = rawurlencode(ob_get_clean());
				
				$label = $this->getParam('share-article');
			}
		
		elseif($plxShow->mode()=='static')
			{				
				/* Récupération de l'URL */
				$url = "?static".$plxShow->staticId();
				$url = $plxShow->plxMotor->urlRewrite($url);
		        	$url = rawurlencode($url);
		        	
				/* Gestion du titre */
				$plxShow->staticTitle();
				$title = ob_get_clean();
				
				$label = $this->getParam('share-page');
			}					

		/* Remplacement du motif {$url} et {$title} par l'URL et le titre de l'article */
		$pattern = array("{\$url}","{\$title}");
		$urlTitle   = array($url, $title);

 
		echo "<div class=\"social-network\"><span class=\"share\">". $label ."</span><ul>";
		
		for  ($i = 1; $i <= $this->getParam('maxItems') ; $i++) 
			{	
				if ($this->getParam($i.'-param2')=="1")
					{
							if (($this->getParam($i.'-param3')) != "" )
								{ 
									echo "<li><a  title=\"".$this->getParam($i.'-param4')."\"  class=\"".$this->getParam($i.'-param5')."\" href=\"".str_replace($pattern, $urlTitle,  $this->getParam($i."-param3"))."\" {$onclick} target=\"_blank\" rel=\"nofollow\" ><span>".$this->getParam($i.'-param4')."</span></a></li>";								
								}    
					}
			}
			
		if ($this->getParam('mail-param2')=="1"){
		echo "<li><a  title=\"".$this->getParam('mail-param4')."\"  class=\"".$this->getParam('mail-param5')."\" href=\"".str_replace($pattern, $urlTitle,  $this->getParam("mail-param3"))."\"  target=\"_blank\" rel=\"nofollow\" ><span>".$this->getParam('mail-param4')."</span></a></li>";								
		}		
						
		echo "</ul></div>";			
	}

	


}

?>
